### PR TITLE
Use node's latest vote for commitment calc. too

### PR DESCRIPTION
**Pull request by ryoqun**
#### Problem

`test_validator_saves_tower` is rigidly testing things (well, while not being brittle, imo:
https://buildkite.com/organizations/anza/analytics/suites/agave/tests/018d9c5f-d789-7dc6-8749-0c423e6080ac?branch=master).

Because of that, just changing commitment level makes it fail....

After some investigation, it turned out there's needless delay for `finalized`.

#### Summary of Changes

Fix the commitment calc, ~hoping nothing is broken, esp our beloved local-cluster tests...~ (EDIT: seems ci has just passed)

### CODE REVIEW

File: core/src/replay_stage.rs [hunk 1]
```diff
@@ -2406,10 +2406,28 @@ impl ReplayStage {
...
+        // It should be acceptable to aggressively use the vote for our own _local view_ of
+        // commitment aggregation, although it's not guaranteed that the new vote transaction is
+        // observed by other nodes at this point.
+        //
+        // The justification stems from the assumption of the sensible voting behavior from the
+        // consensus subsystem. That's because it means there would be a slashing possibility
+        // otherwise.
+        //
+        // This behavior isn't significant normally for mainnet-beta, because staked nodes aren't
+        // servicing RPC requests. However, this eliminates artificial 1-slot delay of the
```
Comment (ryoqun) [inline core/src/replay_stage.rs:2422]: Alternatively, i could just shift assertions of the test by 1 slot. But, I'm somewhat inclined to fixing the impl itself...

Comment (ryoqun) [reply core/src/replay_stage.rs:2422]: i mean, it's odd that root and supermajority root aren't happening at the same time for the single node cluster....

File: core/src/commitment_service.rs [hunk 2]
```diff
@@ -26,14 +26,23 @@ pub struct CommitmentAggregationData {
     bank: Arc<Bank>,
     root: Slot,
     total_stake: Stake,
+    // Node's latest local vote state accompanied to the bank.
+    // Used for commitment aggreagation if the vote account is staked.
```
Comment (CriesofCarrots) [inline core/src/commitment_service.rs]:
```suggestion
    // The latest local vote state of the node running this service.
    // Used for commitment aggregation if the node's vote account is staked.
```

File: core/src/commitment_service.rs [hunk 3]
```diff
@@ -139,8 +148,11 @@ impl AggregateCommitmentService {
         aggregation_data: CommitmentAggregationData,
         ancestors: Vec<u64>,
     ) -> CommitmentSlots {
-        let (block_commitment, rooted_stake) =
-            Self::aggregate_commitment(&ancestors, &aggregation_data.bank);
+        let (block_commitment, rooted_stake) = Self::aggregate_commitment(
+            &ancestors,
+            &aggregation_data.bank,
+            Some(&aggregation_data.node_vote_state),
```
Comment (CriesofCarrots) [inline core/src/commitment_service.rs]: Hmm, `node_vote_state` is never None...? Why bother with the Option at all?

**Review by CriesofCarrots — COMMENTED** [review]
Thanks for tracking this down, @ryoqun ! The fix seems reasonable to me.
Just a question about the type passed to `aggregate_commitment`

File: core/src/commitment_service.rs (hunk 3 above)
Comment (ryoqun) [reply core/src/commitment_service.rs]: i forgot to clean up this before marking this pr out of draft...: https://github.com/anza-xyz/agave/pull/1964/commits/e5c5d95398780f27914aaeeaf9d2067d1824da18

**Review by CriesofCarrots — APPROVED** [review]

**Pull request merged**